File: Python/thread.py
import threading
import time
def thread_task():
    print("Thread {} is running".format(threading.current_thread().name))
    time.sleep(2)
    print("Thread {} is exiting".format(threading.current_thread().name))
    
def main():
    # 创建线程
    thread1 = threading.Thread(target=thread_task, name='Thread 1')
    thread2 = threading.Thread(target=thread_task, name='Thread 2')
    
    # 启动线程
    thread1.start()
    thread2.start()
    
    # 等待线程执行完成
    thread1.join()
    thread2.join()
    
    print("All threads have finished")

if __name__ == "__main__":
    main()

File: Java/thread.java
class MyRunnable implements Runnable {
    private String threadName;
    private int ncycle;
    public MyRunnable(String name) {
        this.threadName = name;
    }

    public void setncycle (int n){
        ncycle=n;
    }
    @Override
    public void run() {
        for (int i = 0; i < ncycle; i++) {
            
        }
    }
}

public class thread {
    int a;
    public static void main(String[] args) {
        Thread thread1 = new Thread(new MyRunnable("线程1"));
        Thread thread2 = new Thread(new MyRunnable("线程2"));
        
        thread1.start(); // 启动线程1
        thread2.start(); // 启动线程2
        
        try {
            thread1.join(); // 等待线程1结束
            thread2.join(); // 等待线程2结束
        } catch (InterruptedException e) {
            System.out.println("主线程被中断!");
        }
        
        System.out.println("所有线程已完成!");
    }
}

